### ISSUE TITLE
std::println() will always be executed in the "if" statement of "for", even if the if is false

### ISSUE THREAD

**Issue by Cc-a23187** [noir-lang/noir#2042]
### Aim

std::println() statement  executed in the "if" statement of "for" loop.

### Expected Behavior

std::println() statement can't be executed in the "if" statement of "for" loop when the "if" statement  is false.

### Bug

![image](https://github.com/noir-lang/noir/assets/33950091/714ac9fc-0d58-4f48-bc48-578e67a275b5)
![image](https://github.com/noir-lang/noir/assets/33950091/fa0d2940-621f-4d1f-9970-1ca847ae1650)
std::println is executed regardless of whether the if control statement is True
In second picture that I let a variable ”temp“. It correctly executes "temp = temp+1" in the third loop.So the if statement block just error while executing std::println().

### To Reproduce

1.
2.
3.
4.

### Installation Method

Binary

### Additional Context

https://discord.com/channels/1113924620781883405/1124022445054111926/1133233367966548008

### Would you like to submit a PR for this Issue?

No

**Comment by kevaundray** [comment]
This is likely with the legacy ssa code, you can confirm this by using the `--experimental-ssa` flag after `--show-output`.

This should be the default in the newer nightly releases, with formatting features coming soon

**Comment by jfecher** [comment]
This is fixed in 0.9.0. The new output for your second example (with x = 1 and y = 1) is:
```
["0x01","0x02","0x00"]
[false,false]
"hello"
"0x01"
["0x02","0x02","0x01"]
[true,false]
["0x03","0x02","0x01"]
[false,false]
```
